openvdb/tools/LevelSetMeasure.h

```diff
@@ -135,77 +138,122 @@ class LevelSetMeasure
     /// @note A grain size of 0 or less disables multi-threading!
     void setGrainSize(int grainsize) { mGrainSize = grainsize; }
 
-    /// @brief Compute the surface area and volume of the level
-    /// set. Use the last argument to specify the result in world or
-    /// voxel units.
-    /// @note This method is faster (about 3x) then the measure method
-    /// below that also computes the average mean-curvature.
-    void measure(Real& area, Real& volume, bool useWorldUnits = true);
-
-    /// @brief Compute the surface area, volume, and average
-    /// mean-curvature of the level set. Use the last argument to
-    /// specify the result in world or voxel units.
-    /// @note This method is slower (about 3x) then the measure method
-    /// above that only computes the area and volume.
-    void measure(Real& area, Real& volume, Real& avgMeanCurvature, bool useWorldUnits = true);
+    /// @brief Compute the surface area of the level set.
+    /// @param useWorldUnits Specifies if the result is in world or voxel units.
+    /// @note Performs internal caching so only the initial call incurs actual computation.
+    Real area(bool useWorldUnits = true);
+
+    /// @brief Compute the volume of the level set surface.
+    /// @param useWorldUnits Specifies if the result is in world or voxel units.
+    /// @note Performs internal caching so only the initial call incurs actual computation.
+    Real volume(bool useWorldUnits = true);
+
+    /// @brief Compute the total mean curvature of the level set surface.
+    /// @param useWorldUnits Specifies if the result is in world or voxel units.
+    /// @note Performs internal caching so only the initial call incurs actual computation.
+    Real totMeanCurvature(bool useWorldUnits = true);
```

Comment:
I think I'd prefer the more explicit "totalMeanCurvature" to the abbreviation of "tot". Likewise "average" instead of "avg".

Reply:
Do "grep avg openvdb/tools/*" and you'll find plenty of hits (all of them code). Now do "grep average openvdb/tools/*" and you get far fewer hits and more importantly none of them are actual code (only documentation). So, in the interest of consistency I will not make the suggested change (just like you pushed back on my recent suggested for a different signature of addChild - using the same argument of inconsistency with existing code :).

Reply:
Consistency is always a good argument. :)

I'm going to approve this, but with a small caveat to your reasoning - most of those hits for "avg" are used in variable names not method names, meaning that they are implementation detail and not public API (implementation detail is less critical IMO). There is in fact only one method name which uses an abbreviated "avg()" which is in Stats.h which also includes an equivalent mean() method. Where possible, I prefer longer but more explicit method names.

Reply:
thanks and yes I agree that in general descriptive names are to be preferred over cryptic abbreviations. I just think that in this case there is a precedence in the code for using avg so at least I would be familiar with (not to say expecting) this naming scheme.